src/client/editor/Editor.js

```diff
@@ -378,6 +378,26 @@ export default class Editor {
     return exportScene(this.scene);
   }
 
+  async extendScene(inheritedURI) {
+    const extendSceneDef = {
+      entities: {},
+      inherits: inheritedURI
+    };
+
+    this.deselect();
+    this._resetHelpers();
+    const ancestors = [];
+    const scene = await loadSerializedScene(extendSceneDef, inheritedURI, this.addComponent, true, ancestors);
+
+    scene.userData._ancestors = ancestors;
```

Comment:
No need to do this, it happens in SceneLoader already.

Reply:
It seems that it does only in `loadScene`, but not `loadSerializedScene`, cuz when I use `loadSerializedScene` without `ancestors` here, the hierarchy panel doesn't display any ancestor files on the top.

Reply:
`loadSerializedScene` calls `loadScene`. You do have to pass an empty array in, but you don't have to set `userData._ancestors` yourself.